tests/src/ArrayAddingIteratorTest.php

```diff
@@ -89,4 +86,27 @@ public function testAddsValueOnlyIfItIsNotPreviouslyStored(): void
             ->from($source);
         $this->assertEquals(1, $called->value());
     }
+
+    /**
+     * @covers ::__construct
+     * @covers ::current
+     * @covers ::key
+     * @covers ::next
+     * @covers ::valid
+     * @covers ::rewind
+     * 
+     * @small
+     *
+     * @return void
+     */
+    public function testReturnsAddedValuesWithIteratorToArray(): void
+    {
+        $input = [4, 3, 94, 25, 63, 6, 72, 7];
+        $this->assertEquals(
+            $input,
+            iterator_to_array(
+                new ArrayAddingIterator([4, 3, 94, 25, 63, 6, 72, 7])
+            )
+        );
+    }
```

Comment:
The solution to the existing problem required assuring that iterator works with `iterator_to_array`.